lib/csv/table.rb

```diff
@@ -258,6 +259,11 @@ def delete(*indexes_or_headers)
             (@mode == :col_or_row and index_or_header.is_a? Integer)
           @table.delete_at(index_or_header)
         else                 # by header
+          if index_or_header.is_a? Integer
+            @headers.delete_at(index_or_header)
+          else
+            @headers.delete(index_or_header)
+          end
```

Comment:
Can you add a test for this case?

Reply:
I've added tests of deleting headers by index and header.